components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java

```diff
@@ -247,20 +338,20 @@ private void validateSecretCreateRequest(Secret secret) throws SecretManagementE
     }
 
     /**
-     * Validate that secret name. Validate the secret existence.
+     * Validate that secret type and secret name is non empty. Validate the secret existence.
      * Set tenant domain if it is not set to the secret object.
      *
      * @param secret The secret to be replaced.
      * @throws SecretManagementException If secret validation fails.
      */
-    private void validateSecretReplaceRequest(Secret secret)
+    private void validateSecretReplaceRequest(String secretTypeName, Secret secret)
             throws SecretManagementException {
 
-        if (StringUtils.isEmpty(secret.getSecretName()) || StringUtils.isEmpty(secret.getSecretValue())) {
+        if (StringUtils.isEmpty(secretTypeName) || StringUtils.isEmpty(secret.getSecretName()) || StringUtils.isEmpty(secret.getSecretValue())) {
```

Comment:
Seems like this has exceeded 120 character limit

Reply:
Addressed with commit [001ee25](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/001ee2501f3eab862efd07a13864e81939cfe648)